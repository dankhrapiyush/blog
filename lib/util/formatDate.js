/**
 * Format time to "M d, Y"
 *
 * @param {string} time
 * @returns {string} A formatted Date string (e.g. `Jul 5, 2020`).
 */
export const formatDate = (time) => {
    return new Date(time).toLocaleDateString(
        undefined,
        {
            year: 'numeric',
            month: 'short',
            day: 'numeric'
        }
    );
};
