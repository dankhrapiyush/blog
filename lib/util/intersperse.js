/**
 * https://stackoverflow.com/a/23619085/1438863
 *
 * intersperse: Return an array with the separator interspersed between
 * each element of the input array.
 *
 * > _([1,2,3]).intersperse(0)
 * [1,0,2,0,3]
 *
 * @param {array} arr
 * @param {string} sep
 * @returns {array}
 */
export const intersperse = (arr, sep) => {
    if (arr.length === 0) {
        return [];
    }

    return arr.slice(1).reduce(
        function (xs, x, i) {
            return xs.concat([sep, x]);
        },
        [arr[0]]
    );
};
