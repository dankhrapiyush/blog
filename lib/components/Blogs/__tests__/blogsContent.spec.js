import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { default as createTestInstance } from '@magento/peregrine/lib/util/createTestInstance';
import BlogsContent from '../blogsContent';
import { IntlProvider } from 'react-intl';

jest.mock('@magento/venia-ui/lib/classify');
jest.mock('@magento/venia-drivers', () => ({
    Link: ({ children, ...rest }) => <div {...rest}>{children}</div>,
    resourceUrl: x => x,
    html: x => x
}));
jest.mock('@magento/venia-ui/lib/components/RichContent', () => 'RichContent');
jest.mock('@magento/venia-ui/lib/components/Pagination', () => 'Pagination');

const blogsContentProps = {
    pageSize: 1,
    data: {
        items: [
            {
                post_id: 1,
                identifier: 'foo',
                title: 'Foo',
                content: 'Foo Description',
                publish_time: '2020-07-05 00:00:00',
                short_content: 'Foo Short Description',
                author: {
                    title: 'web dev',
                    name: 'web dev',
                    url: null,
                    author_url: 'https://magento.local/blog/author/web-dev'
                },
                tags: [
                    {
                        tag_id: 1,
                        identifier: 'tag1',
                        title: 'tag1',
                        is_active: 1
                    }
                ],
                categories: [
                    {
                        category_id: 1,
                        identifier: 'general',
                        title: 'General',
                        is_active: 1
                    }
                ]
            },
            {
                post_id: 2,
                identifier: 'bar',
                title: 'Bar',
                content: 'Bar Description',
                publish_time: '2020-07-10 00:00:00',
                short_content: 'Foo Short Description',
                author: {
                    title: 'web dev',
                    name: 'web dev',
                    url: null,
                    author_url: 'https://magento.local/blog/author/web-dev'
                },
                tags: [
                    {
                        tag_id: 1,
                        identifier: 'tag1',
                        title: 'tag1',
                        is_active: 1
                    },
                    {
                        tag_id: 2,
                        identifier: 'tag2',
                        title: 'tag2',
                        is_active: 1
                    }
                ],
                categories: [
                    {
                        category_id: 1,
                        identifier: 'general',
                        title: 'General',
                        is_active: 1
                    },
                    {
                        category_id: 2,
                        identifier: 'new',
                        title: 'New',
                        is_active: 0
                    }
                ]
            }
        ]
    }
};

test('renders blogs list', () => {
    const root = createTestInstance(
        <MemoryRouter>
            <IntlProvider locale="en-US">
                <BlogsContent {...blogsContentProps} />
            </IntlProvider>
        </MemoryRouter>
    );

    expect(root.toJSON()).toMatchSnapshot();
});
