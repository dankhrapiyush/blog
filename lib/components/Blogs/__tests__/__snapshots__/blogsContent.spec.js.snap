// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders blogs list 1`] = `
Array [
  <ul
    className="postList"
  >
    <li
      className="postHolder"
    >
      <div
        className="postHeader"
      >
        <div
          className="postTitleHolder"
        >
          <h2
            className="postTitle"
          >
            <div
              to="/blog/post/foo"
            >
              Foo
            </div>
          </h2>
        </div>
        <div
          className="postInfo"
        >
          <div
            className="item"
          >
            <span
              className="label"
            >
              <mock-FormattedMessage
                defaultMessage="Posted: "
                id="blogsContent.posted"
              />
            </span>
            <span
              className="value"
            >
              Jul 5, 2020
            </span>
          </div>
          <div
            className="item"
          >
            <span
              className="label"
            >
              <mock-FormattedMessage
                defaultMessage="Categories: "
                id="blogsContent.categories"
              />
            </span>
            <span
              className="value"
            >
              <div
                to="/blog/category/general"
              >
                General
              </div>
            </span>
          </div>
          <div
            className="item"
          >
            <span
              className="label"
            >
              <mock-FormattedMessage
                defaultMessage="Tags: "
                id="blogsContent.tags"
              />
            </span>
            <span
              className="value"
            >
              <div
                to="/blog/tag/tag1"
              >
                tag1
              </div>
            </span>
          </div>
          <div
            className="item"
          >
            <span
              className="label"
            >
              <mock-FormattedMessage
                defaultMessage="Author: "
                id="blogsContent.author"
              />
            </span>
            <span
              className="value"
            >
              web dev
            </span>
          </div>
        </div>
      </div>
      <div
        className="postContent"
      >
        <div
          className="clearfix"
        >
          <div
            className="clearfix"
          >
            <RichContent
              html="Foo Short Description"
            />
          </div>
          <div
            className="postReadMore"
            to="/blog/post/foo"
          >
            <mock-FormattedMessage
              defaultMessage="Read more"
              id="blogsContent.readMore"
            />
             »
          </div>
        </div>
      </div>
    </li>
    <li
      className="postHolder"
    >
      <div
        className="postHeader"
      >
        <div
          className="postTitleHolder"
        >
          <h2
            className="postTitle"
          >
            <div
              to="/blog/post/bar"
            >
              Bar
            </div>
          </h2>
        </div>
        <div
          className="postInfo"
        >
          <div
            className="item"
          >
            <span
              className="label"
            >
              <mock-FormattedMessage
                defaultMessage="Posted: "
                id="blogsContent.posted"
              />
            </span>
            <span
              className="value"
            >
              Jul 10, 2020
            </span>
          </div>
          <div
            className="item"
          >
            <span
              className="label"
            >
              <mock-FormattedMessage
                defaultMessage="Categories: "
                id="blogsContent.categories"
              />
            </span>
            <span
              className="value"
            >
              <div
                to="/blog/category/general"
              >
                General
              </div>
              , 
            </span>
          </div>
          <div
            className="item"
          >
            <span
              className="label"
            >
              <mock-FormattedMessage
                defaultMessage="Tags: "
                id="blogsContent.tags"
              />
            </span>
            <span
              className="value"
            >
              <div
                to="/blog/tag/tag1"
              >
                tag1
              </div>
              , 
              <div
                to="/blog/tag/tag2"
              >
                tag2
              </div>
            </span>
          </div>
          <div
            className="item"
          >
            <span
              className="label"
            >
              <mock-FormattedMessage
                defaultMessage="Author: "
                id="blogsContent.author"
              />
            </span>
            <span
              className="value"
            >
              web dev
            </span>
          </div>
        </div>
      </div>
      <div
        className="postContent"
      >
        <div
          className="clearfix"
        >
          <div
            className="clearfix"
          >
            <RichContent
              html="Foo Short Description"
            />
          </div>
          <div
            className="postReadMore"
            to="/blog/post/bar"
          >
            <mock-FormattedMessage
              defaultMessage="Read more"
              id="blogsContent.readMore"
            />
             »
          </div>
        </div>
      </div>
    </li>
  </ul>,
  <div
    className="pagination"
  >
    <Pagination />
  </div>,
]
`;
